rust/lib/src/lib.rs

```diff
@@ -33,29 +46,32 @@ fn lock_client_return_seed(lightclient: LightClient) -> String {
 
     seed
 }
-fn construct_uri_load_config(
-    uri: String,
-    data_dir: String,
-) -> Result<(zingoconfig::ZingoConfig, u64), String> {
-    let server = construct_server_uri(Some(uri));
-
-    let (mut config, latest_block_height) = match zingolib::load_clientconfig(server, None) {
-        Ok((c, h)) => (c, h),
+fn build_config_from_uri_chaintype(
+    server_uri: String,
+    chain_type: Option<ChainType>,
+) -> zingoconfig::ZingoConfig {
+    let lightwalletd_uri = construct_lightwalletd_uri(Some(server_uri));
+    let chaintype = if *&chain_type.is_some() {
+        chain_type.expect("To unwrap Some(chaintype)")
+    } else {
+        infer_chaintype(&lightwalletd_uri.to_string())
+    };
```

Comment:
Let's get rid of this unwrap:
```suggestion
    let chaintype = match chain_type {
        Some(chaintype) => chaintype,
        None => infer_chaintype(&lightwalletd_uri.to_string()),
    };
```